File: Home Light Automation/workspace/iot-gateway/src/main/java/neu/s/dey/connecteddevices/project/DataUtil.java
package neu.s.dey.connecteddevices.project;

import com.google.gson.Gson;
import neu.s.dey.connecteddevices.project.GeolocationData;;

public class DataUtil {
	
	//method to convert GeolocationData object to JSON
	public String toJsonFromGeolocationData(GeolocationData geolocdata)	{
		String jsonData = null;
		if (geolocdata != null) {
			Gson gson = new Gson();
			jsonData = gson.toJson(geolocdata);
		}
		return jsonData;
	}
	
	//method to convert JSON to GeolocationData object
	public GeolocationData toGeolocationDataFromJson(String jsonData)	{
		GeolocationData geolocdata = null;
		if (jsonData != null && jsonData.trim().length() > 0) {
			Gson gson = new Gson();
			geolocdata = gson.fromJson(jsonData, GeolocationData.class);
		}
		return geolocdata;
	}
	
	
	//method to convert SunriseSunsetData object to JSON
	public String toJsonFromSunriseSunsetData(SunriseSunsetData sunrisesunsetdata)	{
		String jsonData = null;
		if (sunrisesunsetdata != null) {
			Gson gson = new Gson();
			jsonData = gson.toJson(sunrisesunsetdata);
		}
		return jsonData;
	}
	
	//method to convert JSON to SunriseSunsetData object
	public SunriseSunsetData toSunriseSunsetDataFromJson(String jsonData)	{
		SunriseSunsetData sunrisesunsetdata = null;
		if (jsonData != null && jsonData.trim().length() > 0) {
			Gson gson = new Gson();
			sunrisesunsetdata = gson.fromJson(jsonData, SunriseSunsetData.class);
		}
		return sunrisesunsetdata;
	}
}

File: Home Light Automation/workspace/iot-device/apps/project/DataUtil.py
'''
Created on 20-Feb-2020

@author: soman
'''

import json
import logging
from project.GeolocationData import GeolocationData

class DataUtil(object):

    def __init__(self):
        #Format logging and print message
        logging.basicConfig(format='%(asctime)s:%(levelname)s:%(message)s', level=logging.INFO)

    def toJsonFromGeolocationData(self, geolocdata):
        #method to convert GeolocationData to JSON
        geoloc_data={};
        geoloc_data['lat'] = geolocdata.getlat();
        geoloc_data['lng'] = geolocdata.getlng();
        geoloc_data['timezone'] = geolocdata.gettimezone();
        self.jsonSd = json.dumps(geoloc_data)            
        return self.jsonSd
     
    def toGeolocationDataFromJson(self,geolocjsonData, timezonepayload):
        #method to convert JSON to GeolocationData object
        geolocjsonData = str(geolocjsonData).replace("\'", "\"")
        gdDict = json.loads(geolocjsonData)     
        gd = GeolocationData()
        jsonlocation = gdDict['location']
        gd.lat = jsonlocation['lat'] 
        gd.lng = jsonlocation['lng'] 
        print(timezonepayload)
        gd.timezone = timezonepayload["timeZoneId"] 
        return gd